src/Cardano/Wallet/BlockSyncer.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- This module contains the ticking function that is responsible for invoking block acquisition functionality and executing it in periodic fashion.
+--
+-- Known limitations: ticking function makes sure action is not executed on already consumed block, but does not check and handle block gaps (aka catching up)
+
+module Cardano.Wallet.BlockSyncer
+  (
+    BlockHeadersConsumed(..)
+  , tickingFunction
+  ) where
+
+
+import Prelude
+
+import Cardano.Wallet.Primitive
+    ( Block (..), BlockHeader )
+import Control.Concurrent
+    ( threadDelay )
+import Data.Time.Units
+    ( Second, toMicroseconds )
+
+import qualified Data.List as L
+
+
+newtype BlockHeadersConsumed = BlockHeadersConsumed [BlockHeader] deriving (Show, Eq)
+
+storingLimit :: Int
+storingLimit = 2160
+
+tickingFunction
+    :: IO [Block]
+    -- ^ a way to get a new block
+    -> (Block -> IO ())
+    -- ^ action taken on a new block
+    -> Second
+    -- ^ tick time
+    -> BlockHeadersConsumed
+    -> IO ()
+tickingFunction getNextBlocks action tickTime = go
+    where
+      go
+          :: BlockHeadersConsumed
+          -> IO ()
+      go (BlockHeadersConsumed headersConsumed) = do
+          blocksDownloaded <- getNextBlocks
+          let blocksToProcess =
+                  filter (checkIfAlreadyConsumed headersConsumed) blocksDownloaded
+          mapM_ action blocksToProcess
+          threadDelay $ (fromIntegral . toMicroseconds) tickTime
+          go $ BlockHeadersConsumed
+              $ take storingLimit
+              $ map header blocksToProcess ++ headersConsumed
+
+      checkIfAlreadyConsumed
+          :: [BlockHeader]
+          -> Block
+          -> Bool
+      checkIfAlreadyConsumed consumedHeaders (Block theHeader _) =
+          theHeader `L.notElem` consumedHeaders
```

Comment:
Okay, this looks good now. If I may throw a naming suggestion:

- ~~BlockHeaderConsumed~~ ---> `Known` with a type parameter, BlockHeader is rather unnecessary here, it's already in the type: `Known [BlockHeader]`

- ~~checkIfAlreadyConsumed~~ ---> `isKnown`, less prose is good :) 

- ~~tickTime~~ --> `delay`, well, that's what it is in the end. 

- ~~blocksDownloaded~~ --> `blocks`, again, no need to be extra verbose about this. By keeping your function small, the context is enough to understand this. And, in the end, it doesn't really matter if blocks have been _downloaded_, or _fetched_ or _unserialized_ .. Unless it helps disambiguate things, go for a short name.

- ~~`blocksToProcess`~~ --> `block'` or `unknownBlock`, just to be aligned with the above suggestions

- ~~`consumedHeaders`~~ --> `knownHeaders`, also, to be aligned

- ~~`theHeader`~~ --> `h` is probably enough because the function is really short, or, do not pattern match and go for `header block` or `header b` 


Short names are usually good. Too short is sometimes not very helpful, though it make sense when working on abstractions (like in `filter :: (a -> Bool) -> [a] -> [a]`, one wouldn't write `filter :: (theElement -> Bool) -> theList -> theList`).